warehouse/query-core/src/main/java/datawave/query/jexl/JexlASTHelper.java

```diff
@@ -1611,6 +1612,23 @@ public String getFormattedViolations() {
         }
     }
     
+    /**
+     * Checks to see if the tree contains any AND/OR nodes with less than 2 children.
+     * 
+     * @param node
+     *            the tree to validate
+     * @param failHard
+     *            if true, throw a {@link RuntimeException} if a violation was found
+     * @return true if valid, or false otherwise
+     */
+    public static boolean validateJunctionChildren(JexlNode node, boolean failHard) {
```

Comment:
Consider adding a default `valiateJunctionChildren(node)` with a reasonable default value for `failHard`?

Reply:
Easy enough to do, and added.